daemon/status.go

```diff
@@ -115,6 +230,8 @@ func (d *Daemon) collectStatus() {
 		d.statusResponse = response
 		d.statusCollectMutex.Unlock()
 
+		d.setPrometheusMetrics()
```

Comment:
I don't think this is the right place to call this function. defer to @raybejjani

Reply:
Should I create some separate goroutine with an infinite loop which is "sleeping" the same amount of time? I thought that setting Prometheus metrics in the same loop we are refreshing status is fine.

Reply:
The problem is this code is not using the Prometheus client library in a "native" way.  It is taking internal variables and exposing them.  What you want to do here is use the "Const" metric pattern, so that when Prometheus scrapes, the internal data is sent down a channel to the client library.

This is a good example of how to do it:
https://github.com/minio/minio/blob/master/cmd/metrics.go

This other project also had the same pattern, and basically had to re-write the whole thing:
https://github.com/minio/minio/pull/5848

Really, I highly recommend against this pattern, as the Prometheus library itself is very good, thread-safe, and provides simple primitives for exposing metrics like `Inc()` for counters, and `Observe()` for histograms.

Reply:
@SuperQ I already used `Observe()` for histograms. Are you OK with moving Gauges to the separate custom Collector and leaving `Observe()` calls here?

In your example I see that histograms are also "observed" in the way I did, not in the custom Collector.

Reply:
@SuperQ Sorry, disregard my previous comment. I noticed that I can use `MustNewConstHistogram` in my custom collector, so I will try to do that.

Reply:
Or no, const histogram will be worse than normal histogram to us... My first comment is still valid then.